lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -613,14 +616,45 @@ def _delete_boot_diagnostic_container(
 
     def _get_node_information(self, node: Node) -> Dict[str, str]:
         information: Dict[str, Any] = {}
+        for key, method in self._environment_information_hooks.items():
+            node.log.debug(f"detecting {key} ...")
+            try:
+                value = method(node)
+                if value:
+                    information[key] = value
+            except Exception as identifier:
+                node.log.exception(f"error on get {key}.", exc_info=identifier)
+        if node.is_connected and node.is_posix:
+            node.log.debug("detecting lis version...")
+            modinfo = node.tools[Modinfo]
+            information["lis_version"] = modinfo.get_version("hv_vmbus")
+
+            node.log.debug("detecting vm generation...")
+            information[KEY_VM_GENERATION] = node.tools[VmGeneration].get_generation()
+            node.log.debug(f"vm generation: {information[KEY_VM_GENERATION]}")
+
+            information["distro_version"] = node.os.information.full_version
+            information["kernel_version"] = (
+                node.tools[Uname].get_linux_information().kernel_version_raw
+            )
 
-        node.log.debug("detecting lis version...")
-        modinfo = node.tools[Modinfo]
-        information["lis_version"] = modinfo.get_version("hv_vmbus")
+        node_runbook = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        if node_runbook:
+            information["location"] = node_runbook.location
+            information["vmsize"] = node_runbook.vm_size
+            information["image"] = node_runbook.get_image_name()
+        information["platform"] = self.type_name()
+        return information
 
-        node.log.debug("detecting vm generation...")
-        information[KEY_VM_GENERATION] = node.tools[VmGeneration].get_generation()
-        node.log.debug(f"vm generation: {information[KEY_VM_GENERATION]}")
+    @hookimpl
+    def get_node_information(self, node: Node) -> Dict[str, str]:
```

Comment:
It doesn't need to have another layer of wrapper.

Reply:
BTW, where is it registered? It may need to follow the pattern of env info, so it can avoid to be registered multiple times.

Reply:
I didn't add any register, but I saw it was invoked correctly.

Reply:
It may be invoked when call get_environment_information.

Reply:
I set breakpoint, it was invoked from 
```
                    informations: List[
                        Dict[str, str]
                    ] = plugin_manager.hook.get_node_information(node=self._node)
```

Reply:
It sounds not reliable. Did you try on ready or local run? Make sure Azure hook is not invoked.

Reply:
the register happened in platform class.
```
class Platform(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
    def __init__(self, runbook: schema.Platform) -> None:
        super().__init__(runbook)
        self._log = get_logger("", self.type_name())
        plugin_manager.register(self)
```

Reply:
> It sounds not reliable. Did you try on ready or local run? Make sure Azure hook is not invoked.

tried on ready, this hook was not invoked.